### PR TITLE
Adding AbstractProcessor to record base metrics for processors

### PR DESCRIPTION
**Pull request by AustinTag**
*Issue #, if available:*

*Description of changes:*

- Adding AbstractProcessor class
- Using Micrometer to record metrics for records in, records out, and time elapsed in Abstract Processor
- Forced an HdrHistogram version in ES build.gradle since one of the ES dependencies pulls in an old version of this dependency

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kowshikn** [comment]
The latest version makes it easy for the Plugin implementations but I am not sure if this is the best approach we have got. 

* Not sure i we will have issues in the `GET API` if we have prefix based metrics particularly if we use a backend like Promotheus. 
* Will this impact Pipeline-to-pipeline metrics? Any thoughts @yadavcbala ?
* How do we prevent users from not using the standard metrics? I agree this is something implementations should avoid but it would be good if we think of way we can avoid this?

I want to check if using a separate MetricRegistry for each pipeline and also provide each plugin its own registry will make this easier. Also use tags to identify each metric registry.  [Not talking about composite regsitry]

**Comment by AustinTag** [comment]
So the default servlet for Prometheus supports getting metrics for a given list of metric names (but not by prefix). It will not be much extra work us to write an extension to support getting metrics by prefix. Having separate MeterRegistries for each pipeline and plugin will prevent us from using some of the Micrometer functionality, and require us to write a lot of new code to handle this. It is also against the conventions of Micrometer for us to keep many separate registries in the same application. At that point, we may as well not use the face of Micrometer.

File: situp-api/src/main/java/com/amazon/situp/model/metrics/MetricNames.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.amazon.situp.model.metrics;
+
+public class MetricNames {
+    public static final String RECORDS_IN = "recordsIn";
+    public static final String RECORDS_OUT = "recordsOut";
+    public static final String TIME_ELAPSED = "timeElapsed";
+    public static final String DELIMITER = ".";
+}
```
Comment (kowshikn) [inline situp-api/src/main/java/com/amazon/situp/model/metrics/MetricNames.java:8]: nit: Add some documentation for each metrics. I believe these metrics are common for all plugins but some are applicable only to source and some to sink. So lets update them.

File: situp-api/src/main/java/com/amazon/situp/model/processor/AbstractProcessor.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import com.amazon.situp.model.metrics.MetricNames;
+import com.amazon.situp.model.metrics.PluginMetrics;
+import com.amazon.situp.model.record.Record;
+import java.util.Collection;
+import java.util.StringJoiner;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Timer;
+
+public abstract class AbstractProcessor<InputRecord extends Record<?>, OutputRecord extends Record<?>> implements
```
Comment (kowshikn) [inline situp-api/src/main/java/com/amazon/situp/model/processor/AbstractProcessor.java:13]: s/Processor/Piper

File: situp-api/src/main/java/com/amazon/situp/model/processor/AbstractProcessor.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import com.amazon.situp.model.configuration.PluginSetting;
+import com.amazon.situp.model.metrics.MetricNames;
+import com.amazon.situp.model.metrics.PluginMetrics;
+import com.amazon.situp.model.record.Record;
+import java.util.Collection;
+import java.util.StringJoiner;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Timer;
+
```
Comment (kowshikn) [inline situp-api/src/main/java/com/amazon/situp/model/processor/AbstractProcessor.java:12]: nit: Java Doc.

File: situp-api/src/main/java/com/amazon/situp/model/metrics/MetricNames.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.amazon.situp.model.metrics;
+
+public class MetricNames {
```
Comment (kowshikn) [inline situp-api/src/main/java/com/amazon/situp/model/metrics/MetricNames.java:3]: Buffer metrics?

**Review by kowshikn — APPROVED** [review]

File: situp-api/src/main/java/com/amazon/situp/model/metrics/MetricNames.java (hunk 4 above)
Comment (AustinTag) [reply situp-api/src/main/java/com/amazon/situp/model/metrics/MetricNames.java:3]: Sure, will add those when I add the buffer metrics

**Pull request merged**